streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -138,6 +138,7 @@
 
     /** {@code client.id} */
     public static final String CLIENT_ID_CONFIG = CommonClientConfigs.CLIENT_ID_CONFIG;
+    private static final String CLIENT_ID_DOC = "An id string, which will be used as a prefix for internal consumer, producer and restore-consumer";
```

Comment:
Agreed.. This description seems perfect. 
I don't know how I missed StreamThread in first place :| .. I am just starting up with kafka code.. But this should have been caught in find usage.. Anyways.. 

I have updated PR with the updated description string. Please check if it looks good.